tests/unit/assessment/conftest.py

```diff
@@ -0,0 +1,236 @@
+            'Limited Evidence',
+            'No Evidence'
+        ])
+
+    yield gen_random_grade
+
+
+@pytest.fixture
+def code_generator(randomstring):
+    """ Creates some example code to use.
+    """
+
+    def gen_code():
+        return f"""
+    function x{randomstring(10)} () {{
+        var v{randomstring(10)} = {random.randint(0, 999)};
+    }}
+    """
+
+    yield gen_code
+
+
+@pytest.fixture
+def code(code_generator):
+    yield code_generator()
+
+
+@pytest.fixture
+def prompt():
+    """ Creates a generic prompt.
+    """
+
+    yield """
+You are a teaching assistant for a classroom of kids ages 5 to 15.
+You are teaching computer science.
+"""
+
+@pytest.fixture
+def openai_api_key(randomstring):
+    """ Creates a random but reasonable OpenAI API key.
+    """
+
+    yield f"sk-{randomstring(48)}"
+
+
+@pytest.fixture
+def rubric(randomstring):
+    """ Creates a generic rubric (a CSV string).
+    """
+
+    output = io.StringIO()
+    writer = csv.writer(output, csv.QUOTE_NONNUMERIC)
+
+    # Header
+    writer.writerow(['Key Concept', 'Extensive Evidence', 'Convincing Evidence', 'Limited Evidence', 'No Evidence'])
+
+    for i in range(0, random.randint(2, 5)):
+        key_concept = f"Concept {randomstring(10)}"
+        writer.writerow([key_concept, randomstring(20), randomstring(23), randomstring(25), randomstring(15)])
+
+    yield output.getvalue()
+
+
+@pytest.fixture
+def example_generator(code, randomstring, random_grade_generator):
+    """ Creates a random example.
+    """
+
+    def gen_example(rubric):
+        # Generate the example grades (which are TSV formatted)
+        example_rubric = "Key Concept\tObservations\tGrade\tReason\n"
+
+        # Generate a grade for each concept
+        parsed_rubric = list(csv.DictReader(rubric.splitlines()))
+        example_rubric += '\n'.join(
+            map(
+                lambda key_concept: f'{key_concept}\t{randomstring(10)}\t{random_grade_generator()}\t{randomstring(12)}',
+                set(x['Key Concept'] for x in parsed_rubric)
+            )
+        )
+
+        # Return the listing
+        return [
+            code,
+            example_rubric
+        ]
+
+    yield gen_example
+
+
+@pytest.fixture
+def examples(example_generator):
+    """ Creates a random set of examples to pass into an assessment function.
+    """
+
+    def gen_examples(rubric, num_examples=None):
+        if num_examples is None:
+            num_examples = random.randint(1,  5)
+
+        return list(
+            map(
+                lambda x: example_generator(rubric),
+                range(0, num_examples)
+            )
+        )
+    
+    yield gen_examples
+
+
+@pytest.fixture
+def llm_model(randomstring):
+    """ Creates a resonable looking llm_model tag.
+    """
+
+    yield f"gpt-{randomstring(10)}"
+
+
+@pytest.fixture
+def num_responses():
+    """ Creates a valid num_responses value.
+    """
+
+    yield random.randint(1, 3)
```

Comment:
This should certainly be randint(2, 3) ... I believe the n=1 codepath is already tested separately, but perhaps not sufficiently. it's still covered with the change to the lower bound to 2.

Reply:
Yes, it more or less is tested because I seem to often generate GPT responses that only have 1 choice. And then the main test is split across the random num_responses and just 1... so that confirms that the lower bound is better as 2.